### ISSUE TITLE
Incorrect selected option returned if multiple options share a value

### ISSUE THREAD

**Issue by SyntheticDave** [ivaynberg/select2#2894]
I have a select element with multiple optgroups, where it is possible for options in different optgroups to have the same value. For example (see the 'Apple' and 'Potato' options):

```html
<select data-placeholder="Fruit/Veg" id="food_select">
  <option value=""></option>

  <optgroup label="Fruits">
    <option data-params="{'selected_type':'Fruit'}" value="1">Apple</option>
    <option data-params="{'selected_type':'Fruit'}" value="2">Pear</option>
    <option data-params="{'selected_type':'Fruit'}" value="3">Banana</option>
  </optgroup>

  <optgroup label="Veg">
    <option data-params="{'selected_type':'Veg'}" value="1">Potato</option>
    <option data-params="{'selected_type':'Veg'}" value="4">Cabbage</option>
    <option data-params="{'selected_type':'Veg'}" value="5">Bean</option>
  </optgroup>
</select>
```

With a plain select, attempting to find the selected option with jQuery yields the correct result. `$('#food_select').find(:selected)`

However, when select2 is applied, and 'Potato' is selected, `$('#food_select').find(:selected)` returns the 'Apple' option, which is the first option with the value of '1'.

All other selected options work correctly, so it seems select2 is not handling the duplicate values correctly. I've created a JSFiddle demonstrating this behaviour, at: http://jsfiddle.net/e57m9cax/.

Is there a workaround or fix for this behaviour?

I've also posted this as a question on StackOverflow: http://stackoverflow.com/questions/27772542/select2-incorrect-selected-option-returned-if-multiple-options-share-a-value.

Thanks.

**Comment by kevin-brown** [comment]
This is a known issue and because of how the internals are structured for 3.5, it cannot be fixed in the 3.x released.

Luckily it's fixed in 4.0, which will be entering into the first beta shortly: https://github.com/ivaynberg/select2/pull/2743

**Comment by SyntheticDave** [comment]
Thanks for the quick response, I guess I'll just use separate selects for now.
Looking forward to the release of 4.0.
